google/cloud/common_options.h

```diff
@@ -227,12 +228,22 @@ struct EnableServerRetriesOption {
   using Type = bool;
 };
 
+/**
+ * An option to inject custom headers into the request.
+ *
+ * For REST endpoints, these headers are added to the HTTP headers. For gRPC
+ * endpoints, these headers are added to the `grpc::ClientContext` metadata.
```

Comment:
add: `@ingroup options` which I am noticing I forgot to do for `EnableServerRetriesOption`.